### PR TITLE
Exposes verticalScrollContainer.igxForOf in a public method

### LINKED ISSUE #5069: [igx-grid] expose grid.verticalScrollContainer.igxForOf as dataView property
## Is your feature request related to a problem? Please describe.
Before, I asked you to expose `filteredSotedData` property as a public API. Because I wanted to get dataView.

https://github.com/IgniteUI/igniteui-angular/issues/1432

`filteredSotedData` made my demand at that time . But it is not enough now because`igx-grid` has grouping feature and then `filteredSotedData`  is not equal to the dataView when grouping.

Instead, `grid.verticalScrollContainer.igxForOf` is exactly the dataView what I want.

## Describe the solution you'd like
I'd like you to expose `grid.verticalScrollContainer.igxForOf` as `dataView` property like this.
```
// grid-base.component.ts
get dataView(): any[] {
  return this.grid.verticalScrollContainer.igxForOf;
}

// how to use
const dataView = this.grid.dataView;

```
## Describe alternatives you've considered
I have no choice but to use `this.grid.verticalScrollContainer.igxForOf` directly. But it is not a public API(I thought it was not marked as hidden before...).

https://github.com/IgniteUI/igniteui-angular/blob/7.3.4/projects/igniteui-angular/src/lib/grids/grid-base.component.ts#L1715-L1719

## Additional context
None.

### PR DESCRIPTION
**Pull request by 3phase**
Closes #5069   

### Additional information (check all that apply):
 - [x] Bug fix
 - [ ] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [x] All relevant tags have been applied to this PR
 - [ ] This PR includes unit tests covering all the new code
 - [ ] This PR includes API docs for newly added methods/properties
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [x] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

File: projects/igniteui-angular/src/lib/grids/common/grid.interface.ts [hunk 1]
```diff
@@ -24,6 +24,7 @@ export interface GridType {
     pinnedColumns: any;
     summariesRowList: any;
     headerContainer: any;
+    dataView: any;
```
Comment (MayaKirova) [inline projects/igniteui-angular/src/lib/grids/common/grid.interface.ts]: Type should be array: any[]

File: CHANGELOG.md [hunk 2]
```diff
@@ -75,6 +75,7 @@ For more information about the theming please read our [documentation](https://w
     - `isCellSelected` method has been deprecated. Now you can use `selected` property.
     - `rowSelectable` property has been deprecated. Now you can use `rowSelection` property to enable row selection and also you can show and hide the row selectors by setting `hideRowSelectors` property to true or false (which is the default value).
     - Removed deprecated event `OnFocusChange`
+    - Privately accessed method, `verticalScrollContainer.igxForOf`, is now publicly exposed as `dataView`
```
Comment (MayaKirova) [inline CHANGELOG.md]: Describe this as adding a new public property that returns the current transformed paged/filtered/sorted/grouped data that is displayed in the grid.

File: projects/igniteui-angular/src/lib/grids/grid-base.component.ts [hunk 3]
```diff
@@ -4997,6 +4997,17 @@ export abstract class IgxGridBaseComponent extends DisplayDensityBase implements
         return 0;
     }
 
+    /**
+     * Returns a complete list of the elements in the current vertical dataview.
```
Comment (MayaKirova) [inline projects/igniteui-angular/src/lib/grids/grid-base.component.ts]: Returns the current transformed paged/filtered/sorted/grouped data that is displayed in the grid.

Comment (3phase) [reply projects/igniteui-angular/src/lib/grids/grid-base.component.ts]: Requested changes resolved

**Comment by 3phase** [comment]
> verticalScrollContainer.igxForOf is also used in many places in the navigation services. It should be changed to use dataView as well for consistency.

I can't find any other uses than the already changed ones

File: CHANGELOG.md [hunk 4]
```diff
@@ -75,6 +75,7 @@ For more information about the theming please read our [documentation](https://w
     - `isCellSelected` method has been deprecated. Now you can use `selected` property.
     - `rowSelectable` property has been deprecated. Now you can use `rowSelection` property to enable row selection and also you can show and hide the row selectors by setting `hideRowSelectors` property to true or false (which is the default value).
     - Removed deprecated event `OnFocusChange`
+    - `IgxGridBaseComponent` exposes a new method, `dataView` that returns the currently transformed paged/filtered/sorted/grouped data, displayed in the grid
```
Comment (MayaKirova) [inline CHANGELOG.md]: It's currently a property not a method.

File: projects/igniteui-angular/src/lib/grids/grid-base.component.ts [hunk 5]
```diff
@@ -4997,6 +4997,17 @@ export abstract class IgxGridBaseComponent extends DisplayDensityBase implements
...
     }
 
+    /**
+     * Returns the currently transformed paged/filtered/sorted/grouped data, displayed in the grid.
+     * ```typescript
+     *      const dataView = this.grid.dataView;
+     * ```
+     * @memberof IgxGridComponent
+     */
+    get dataView(): any {
```
Comment (MayaKirova) [inline projects/igniteui-angular/src/lib/grids/grid-base.component.ts]: Type is any[]

**Pull request merged**